src/ValueParsers/DateFormatParser.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+namespace ValueParsers;
+
+use DataValues\TimeValue;
+
+/**
+ * This parser is in essence the inverse operation of Language::sprintfDate.
+ *
+ * @see Language::sprintfDate
+ *
+ * @since 0.8.1
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class DateFormatParser extends StringValueParser {
+
+	const FORMAT_NAME = 'datetime';
+
+	const OPT_DATE_FORMAT = 'dateFormat';
+	const OPT_DIGIT_TRANSFORM_TABLE = 'digitTransformTable';
+
+	/**
+	 * Must be a two-dimensional array, the first dimension mapping the month's numbers 1 to 12 to
+	 * arrays of month
```

Comment:
"Should be [...] to arrays of localized month names, possibly including abbreviated and genitive month names. Can also be a one-dimensional array of strings."